crates/router/src/core/payment_link.rs

```diff
@@ -287,16 +276,95 @@ fn validate_order_details(
     Ok(updated_order_details)
 }
 
-fn extract_payment_link_config(
-    pl_config: Option<serde_json::Value>,
-) -> Result<Option<admin_types::PaymentLinkConfig>, error_stack::Report<errors::ApiErrorResponse>> {
-    pl_config
-        .map(|config| {
-            serde_json::from_value::<admin_types::PaymentLinkConfig>(config)
-                .into_report()
-                .change_context(errors::ApiErrorResponse::InvalidDataValue {
-                    field_name: "payment_link_config",
-                })
+pub fn extract_business_payment_link_config(
+    pl_config: serde_json::Value,
+) -> Result<admin_types::BusinessPaymentLinkConfig, error_stack::Report<errors::ApiErrorResponse>> {
+    serde_json::from_value::<admin_types::BusinessPaymentLinkConfig>(pl_config.clone())
+        .into_report()
+        .change_context(errors::ApiErrorResponse::InvalidDataValue {
+            field_name: "payment_link_config",
+        })
+}
+
+pub fn extract_payment_link_config(
+    pl_config: serde_json::Value,
+) -> Result<api_models::admin::PaymentLinkConfig, error_stack::Report<errors::ApiErrorResponse>> {
+    serde_json::from_value::<api_models::admin::PaymentLinkConfig>(pl_config.clone())
+        .into_report()
+        .change_context(errors::ApiErrorResponse::InvalidDataValue {
+            field_name: "payment_link_config",
+        })
+}
```

Comment:
Unless there's a specific requirement, we don't need separate functions to handle serde parsing. Search for `.parse_value()`